engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/DataMigrationUtility.java

```diff
@@ -87,19 +92,22 @@
      *  the migration is terminated.
      */
     private boolean filesReadyToMigrate(Long srcDataStoreId) {
-        String[] validStates = new String[]{"Ready", "Allocated", "Destroying", "Destroyed", "Failed"};
+        State[] validStates = {State.Ready, State.Allocated, State.Destroying, State.Destroyed, State.Failed};
         boolean isReady = true;
         List<TemplateDataStoreVO> templates = templateDataStoreDao.listByStoreId(srcDataStoreId);
         for (TemplateDataStoreVO template : templates) {
-            isReady &= (Arrays.asList(validStates).contains(template.getState().toString()));
+            isReady &= (Arrays.asList(validStates).contains(template.getState()));
+            LOGGER.trace(String.format("template state: ", template.getState()));
```

Comment:
```suggestion
            LOGGER.trace(String.format("template state: %s", template.getState().toString()));
```

Reply:
this is automatic @sureshanaparti, the State has  toString() method.

Reply:
the state is not logged in the trace, check other similar logs as well.

Reply:
:facepalm: you are right :shootingmyslefinthehead:

Reply:
```suggestion
            LOGGER.trace(String.format("template state: %s", template.getState()));
```